src/main/java/com/bootcamp/demo/controller/PaymentController.java

```diff
@@ -0,0 +1,36 @@
+package com.bootcamp.demo.controller;
+
+import com.bootcamp.demo.exception.PaymentFailException;
+import com.bootcamp.demo.model.Card;
+import com.bootcamp.demo.model.Transaction;
+import com.bootcamp.demo.service.PaymentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+
+@RestController
+@RequestMapping("/payment")
+public class PaymentController {
+
+    private final PaymentService paymentService;
+
+    @Autowired
+    public PaymentController(PaymentService paymentService) {
+        this.paymentService = paymentService;
+    }
+
+    @GetMapping("/test")
+    public Transaction getTestTransaction(){
+        Transaction transaction = new Transaction("123435", "657353845", 234.43,  LocalDateTime.now());
+        return transaction;
+    }
+
+    @PostMapping("/pay")
```

Comment:
You could add the `consumes = APPLICATION_JSON_VALUE` (from `org.springframework.util.MimeTypeUtils`) parameter to this `@PostMapping`. Alongside with the "produces" parameter specified on the RequstMapping you clearly define the request / response body type.

Reply:
Updated base branch to feature/payment-by-card instead of master. Those are not my modifications